### PR TITLE
[Mobile] Reduced ICU files for HybridGlobalization 

### PR DESCRIPTION
**Pull request by mkhamoyan**
First version of reduced ICU files for HybridGlobalization, Mobile. Contributes to https://github.com/dotnet/runtime/issues/80689.

HybridGlobalization is an optional function that can be switched on by the developer by adding

`<HybridGlobalization>true</HybridGlobalization>`

to the ios/tvos/maccatalyst project.

By this PR `icudt_hg_mobile.json` is added. In comparison with `icudt_mobile.json` below resources are filtered out from `locales_tree`
```
  "-/NumberElements/*/symbols",
  "-/calendar/Gregorian/AmPmMarkers",
  "-/calendar/Gregorian/AmPmMarkersAbbr",
  "-/calendar/Gregorian/AmPmMarkersNarrow",
```
Below are the results comparing current full icu file size with new file size.
`icudt_hybrid.dat `will be created for `iOS/tvOS/macCatalyst` platforms.

| Target arch  | icudt.dat| icudt_hybrid.dat |
| ------------- | ------------- | ------------- |
| maccatalyst x64 | 1 998 352 bytes | 1 983 408 bytes  |
| |   |   |

File size is decreased with ~1%.

cc @SamMonoRT

### CODE REVIEW

File: icu-filters/icudt_hg_mobile.json [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+                "-/NumberElements/latn/miscPatterns",
+                "-/NumberElements/latn/patternsLong",
+                "-/NumberElements/latn/patternsShort",
+                "-/NumberElements/*/patternsLong",
+                "-/NumberElements/*/patternsShort",
+                "-/NumberElements/minimalPairs",
+                "-/NumberElements/*/symbols",
+                "-/calendar/Gregorian/AmPmMarkers",
+                "-/calendar/Gregorian/AmPmMarkersAbbr",
+                "-/calendar/Gregorian/AmPmMarkersNarrow",
```
Comment (ilonatommy) [inline icu-filters/icudt_hg_mobile.json:254]: These 4 exclusions are the only change in comparison to `icudt_mobile.json`? Maybe let's include this info in the PR description.

**Review by ilonatommy — APPROVED** [review]

File: icu-filters/icudt_hg_mobile.json (hunk 1 above)
Comment (mkhamoyan) [reply icu-filters/icudt_hg_mobile.json:254]: Updated the description.

**Comment by mkhamoyan** [comment]
@directhex could you please build and update eng/prebuilts/mobile/icudt_hybrid.dat ? In my machine each time it creates different size dat file from 2 MB to 2.5 MB

**Comment by mkhamoyan** [comment]
As discussed with @directhex , pushed icudt_hybrid.dat size is the same as in his locale build.

**Pull request merged**